terraform/warehouse/vcl/main.vcl

```diff
@@ -259,6 +259,33 @@ sub vcl_recv {
     # Pass the client IP address back to the backend.
     if (req.http.Fastly-Client-IP) {
         set req.http.Warehouse-IP = req.http.Fastly-Client-IP;
+        # Salt & hash the Client IP address
+        declare local var.salt STRING;
+        declare local var.client_ip STRING;
+        declare local var.hashed_ip STRING;
+        # TODO: Can't use Config/Secret stores in VCL, should we inject from TF?
+        set var.salt = "halloumi";
+        # Concatenate the client IP address and the salt
+        set var.client_ip = req.http.Fastly-Client-IP + var.salt;
+        # Hash the concatenated string
+        set var.hashed_ip = digest.hash_sha256(var.client_ip);
+        # TODO: After backend is updated, replace or remove `req.http.Warehouse-IP` above
+        set req.http.Warehouse-Hashed-IP = var.hashed_ip;
+
+        # Geolocate the client IP address. We want details like city, state, country, etc.
+        declare local var.geo STRING;
+        if (client.geo.country_code != "**") {
+            set var.geo = client.geo.country_name;
+            if (client.geo.region != "NO REGION" && client.geo.region != "?") {
+            set var.geo = var.geo + "-" + client.geo.region;
+            }
+            # add city if it's not the same as the region
+            if (client.geo.city != "NO CITY" && client.geo.city != "?"
+                && client.geo.city != client.geo.region) {
+            set var.geo = var.geo + "-" + client.geo.city;
+            }
+        }
+        set req.http.Warehouse-Geo = var.geo;
```

Comment:
should we split these into separate headers or encode them using something like json?

Main reason I ask is I don't think it's possible to reverse this encoding to get the country/region/city back out if any of them contain a `-` character.

Reply:
Given that we might not want to display them in exactly the same format, +1 to providing them in a more structured way

Reply:
Well even for display, if we want to display them as separate fields we would need to split them back into separate fields yea? Like on GitHub when I view my sessions for geolocation data, I see something like:

![image](https://user-images.githubusercontent.com/145979/229542465-e91cd2d7-4e0f-4ac8-ad9d-c8d7eca7748b.png)

or 

![image](https://user-images.githubusercontent.com/145979/229542742-629771f5-1358-41b4-8a40-5afca764efca.png)

This would hard code us to only being able to (reliably) display in COUNTRY-REGION-CITY.

Reply:
Addressed in subsequent commit.